ReGoap/Planner/ReGoapPlanner.cs

```diff
@@ -70,7 +70,7 @@ public ReGoapPlanner(ReGoapPlannerSettings settings = null)
 
                 goalState = goalState.Clone();
                 var leaf = (ReGoapNode<T, W>)astar.Run(
-                    ReGoapNode<T, W>.Instantiate(this, goalState, null, null), goalState, settings.MaxIterations, settings.PlanningEarlyExit);
+                    ReGoapNode<T, W>.Instantiate(this, goalState, null, null), goalState, settings.MaxIterations, settings.PlanningEarlyExit, debugPlan : agent.debugPlan);
```

Comment:
I would put the debugPlan inside the Planner settings (where MaxIterations is set) and call it something more sensible, ex. "graphvizDebug"